src/main/java/pl/psnc/dei/service/TranscriptionPlatformService.java

```diff
@@ -208,8 +213,26 @@ public JsonObject fetchTranscriptionUpdate(Transcription transcription) {
 					}
 				})
 				.block();
+		return JSON.parse(retrieveCurrentTranscription(transcription, response));
+	}
 
-		return JSON.parse(response);
+	private String retrieveCurrentTranscription(Transcription transcription, String enrichments){
+		String transcriptionId = transcription.getTp_id();
```

Comment:
Why don't you pass this id as parameter instead of the whole transcription object? This id is the only thing needed from transcription inside this method...